### PR TITLE
Wrote docs for unhappy paths

### PR DESCRIPTION
**Pull request by twerkmeister**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/llms/unhappy-paths.mdx [hunk 1]
```diff
@@ -11,14 +11,150 @@ import RasaDiscoveryBanner from "@theme/RasaDiscoveryBanner";
...
+
+In the above conversation you can see that
+* The transferring money flow was interrupted by the checking balance flow and the needed information was provided
+* The bot commented on returning to the original flow
+* The bot restated the current question of the original flow
+
+This represents the default behavior in flows. You can adjust the behavior by altering
+the default flow called `pattern_continue_interruped` which is specified and provided
+by default as
+```yaml
```
Comment (m-vdb) [inline docs/docs/llms/unhappy-paths.mdx:49]:
```suggestion
```yaml title="domain.yml"
```

File: docs/docs/llms/unhappy-paths.mdx [hunk 2]
```diff
@@ -11,14 +11,150 @@ import RasaDiscoveryBanner from "@theme/RasaDiscoveryBanner";
...
+        set_slots:
+          - rasa_corrected_slots: None
+        next: "3"
+      - id: "3"
+        action: utter_not_corrected_previous_input
+
+```
+By overriding these defaults, you could, for example, either remove the confirmation
+for correcting slots or make correcting slots impossible. You can override these defaults
+by defining flows, slots, or responses with the respective names shown here.
```
Comment (m-vdb) [inline docs/docs/llms/unhappy-paths.mdx:157]:
```suggestion
You can override these defaults by defining flows, slots, or responses with the
respective names shown here. For instance, you could either remove the confirmation for correcting slots or disable slots correction altogether.
```

**Review by m-vdb — COMMENTED** [review]
great stuff! thanks @twerkmeister

File: docs/docs/llms/unhappy-paths.mdx [hunk 3]
```diff
@@ -11,14 +11,150 @@ import RasaDiscoveryBanner from "@theme/RasaDiscoveryBanner";
...
+What you cannot see from this conversation is that the flow graph was also retraced in
+the background starting from the altered recipient question to make sure the updated
+information is still in line with any logical checks along the flow graph. If the update
+had resulted in going down a different path we would see the questions and utterances
+of this alternate path being used.
+
+the above represents the default behavior in flows. You can adjust the behavior by altering
+the default flow called `pattern_correction` which is specified and provided
+by default as:
+```yaml
```
[the remark this answers is not in the document]
Comment (twerkmeister) [reply docs/docs/llms/unhappy-paths.mdx:99]: see above

File: docs/docs/llms/unhappy-paths.mdx [hunk 4]
```diff
@@ -11,14 +11,150 @@ import RasaDiscoveryBanner from "@theme/RasaDiscoveryBanner";
...
+## Flows with Unhappy Paths
 
-- default flows are in https://github.com/RasaHQ/rasa/pull/12501/files#diff-ef650ec4c31fc5423a8811c54ad1abed3c2c8501b6640fb782144a725e362053
+A flow captures a specific business function as a directed graph.
+These graphs can be purely linear, or branch heavily, or have cycles in certain sections.
+To limit complexity for flow creators, we want them to be able to focus on the direct
+or "happy path" where the user sticks to their original goal and where these goals are
+attained step by step. Instead of having to model flow switches, cancellations, or
+slot corrections explicitly, rasa handles these things in a uniform manner across flows,
+greatly reducing the modeling complexity.
```
[the remark this answers is not in the document]
Comment (twerkmeister) [reply docs/docs/llms/unhappy-paths.mdx]: done outside github bc there was still a "sticks" too much in the suggestion

**Pull request merged**